src/server/game/Entities/Player/PlayerUpdates.cpp

```diff
@@ -0,0 +1,2371 @@
+#include "Pet.h"
+#include "Player.h"
+#include "SavingSystem.h"
+#include "ScriptMgr.h"
+#include "SkillDiscovery.h"
+#include "SpellAuraEffects.h"
+#include "UpdateFieldFlags.h"
+#include "Vehicle.h"
+#include "WeatherMgr.h"
+
+// Zone Interval should be 1 second
+constexpr auto ZONE_UPDATE_INTERVAL = 1000;
+
+constexpr auto CINEMATIC_UPDATEDIFF = 500;
+constexpr auto CINEMATIC_LOOKAHEAD  = 2000;
+
+void Player::Update(uint32 p_time)
+{
+    if (!IsInWorld())
+    {
+        return;
+    }
+
+    sScriptMgr->OnBeforePlayerUpdate(this, p_time);
+
+    // undelivered mail
+    if (m_nextMailDelivereTime && m_nextMailDelivereTime <= time(nullptr))
+    {
+        SendNewMail();
+        ++unReadMails;
+
+        // It will be recalculate at mailbox open (for unReadMails important
+        // non-0 until mailbox open, it also will be recalculated)
+        m_nextMailDelivereTime = time_t(0);
+    }
+
+    // Update cinematic location, if 500ms have passed and we're doing a
+    // cinematic now.
+    m_cinematicDiff += p_time;
+    if (m_cinematicCamera && m_activeCinematicCameraId &&
+        GetMSTimeDiffToNow(m_lastCinematicCheck) > CINEMATIC_UPDATEDIFF)
+    {
+        m_lastCinematicCheck = getMSTime();
+        UpdateCinematicLocation(p_time);
+    }
+
+    // used to implement delayed far teleports
+    SetMustDelayTeleport(true);
+    Unit::Update(p_time);
+    SetMustDelayTeleport(false);
+
+    time_t now = time(nullptr);
+
+    UpdatePvPFlag(now);
+    UpdateFFAPvPFlag(now);
+
+    UpdateContestedPvP(p_time);
+
+    UpdateDuelFlag(now);
+
+    CheckDuelDistance(now);
+
+    UpdateAfkReport(now);
+
+    // Xinef: update charm AI only if we are controlled by creature or
+    // non-posses player charm
+    if (IsCharmed() && !HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_POSSESSED))
+    {
+        m_charmUpdateTimer += p_time;
+        if (m_charmUpdateTimer >= 1000)
+        {
+            m_charmUpdateTimer = 0;
+            if (Unit* charmer = GetCharmer())
+                if (charmer->IsAlive())
+                    UpdateCharmedAI();
+        }
+    }
+
+    if (now > m_Last_tick)
+    {
+        // Update items that have just a limited lifetime
+        UpdateItemDuration(uint32(now - m_Last_tick));
+
+        // check every minute, less chance to crash and wont break anything.
+        UpdateSoulboundTradeItems();
+
+        // Played time
+        uint32 elapsed = uint32(now - m_Last_tick);
+        m_Played_time[PLAYED_TIME_TOTAL] += elapsed; // Total played time
+        m_Played_time[PLAYED_TIME_LEVEL] += elapsed; // Level played time
+        GetSession()->SetTotalTime(GetSession()->GetTotalTime() + elapsed);
+        m_Last_tick = now;
+    }
+
+    // If mute expired, remove it from the DB
+    if (GetSession()->m_muteTime && GetSession()->m_muteTime < now)
+    {
+        GetSession()->m_muteTime = 0;
+        LoginDatabasePreparedStatement* stmt =
+            LoginDatabase.GetPreparedStatement(LOGIN_UPD_MUTE_TIME);
```

Comment:
Can 1 line

Reply:
im gonna look on how to fix the line breaking, hm.